src/app/shared/actions/volume-actions/volume-resize/volume-resize.component.html

```diff
@@ -9,29 +9,17 @@ <h3 class="mat-dialog-title">
     novalidate
   >
     <div class="mat-dialog-content">
-      <div *ngIf="!volumeIsRoot">
-        <mat-form-field
-          class="full-width-input"
-        >
-          <mat-select
-            class="form-select-control"
-            name="diskOffering"
-            [(ngModel)]="diskOfferingId"
-            [placeholder]="'VM_PAGE.STORAGE_DETAILS.VOLUME_RESIZE.DISK_OFFERING' | translate"
-            required
-            (selectionChange)="updateDiskOffering($event.value)"
-          >
-            <mat-option
-              *ngFor="let offering of diskOfferings"
-              [value]="offering.id"
-            >
-              {{ offering.name }}
-            </mat-option>
-          </mat-select>
-        </mat-form-field>
+      <div *ngIf="!volumeIsRoot && diskOffering">
+        <cs-disk-offering-selector
+          name="diskOffering"
+          [diskOfferings]="diskOfferings"
+          [params]="diskOfferingParams"
+          [ngModel]="diskOffering"
+          (change)="updateDiskOffering($event)"
+        ></cs-disk-offering-selector>
       </div>
       <cs-slider
-        *ngIf="isCustomized(diskOfferingId) || volumeIsRoot"
+        *ngIf="isCustomized(diskOffering?.id) || volumeIsRoot"
```

Comment:
We can use diskOffering as input and check its isCustomized property instead of searching item in the array and then checking that item.